v2/jdbc-to-spanner/src/main/java/com/google/cloud/teleport/v2/templates/JdbcToSpanner.java

```diff
@@ -0,0 +1,168 @@
+      "The JDBC drivers for the relational database must be available.",
+      "The Spanner tables must exist before pipeline execution.",
+      "The Spanner tables must have a compatible schema.",
+      "The relational database must be accessible from the subnet where Dataflow runs."
+    })
+public class JdbcToSpanner {
+
+  /**
+   * Main entry point for executing the pipeline. This will run the pipeline asynchronously. If
+   * blocking execution is required, use the {@link JdbcToSpanner#run} method to start the pipeline
+   * and invoke {@code result.waitUntilFinish()} on the {@link PipelineResult}.
+   *
+   * @param args The command-line arguments to the pipeline.
+   */
+  public static void main(String[] args) {
+    UncaughtExceptionLogger.register();
+
+    // Parse the user options passed from the command-line
+    JdbcToSpannerOptions options =
+        PipelineOptionsFactory.fromArgs(args).withValidation().as(JdbcToSpannerOptions.class);
+
+    run(options);
+  }
+
+  /**
+   * Create the pipeline with the supplied options.
+   *
+   * @param options The execution parameters to the pipeline.
+   * @return The result of the pipeline execution.
+   */
+  @VisibleForTesting
+  static PipelineResult run(JdbcToSpannerOptions options) {
+    Pipeline pipeline = Pipeline.create(options);
+    Map<String, Set<String>> columnsToIgnore = getColumnsToIgnore(options);
+    for (String table : getTables(options)) {
+      PCollection<Mutation> rows =
+          pipeline.apply(
+              "ReadPartitions_" + table, getJdbcReader(table, columnsToIgnore.get(table), options));
+      rows.apply("Write_" + table, getSpannerWrite(options));
+    }
+    return pipeline.run();
+  }
+
+  private static List<String> getTables(JdbcToSpannerOptions options) {
+    return Arrays.asList(options.getTables().split(","));
+  }
+
+  private static Map<String, Set<String>> getColumnsToIgnore(JdbcToSpannerOptions options) {
+    String ignoreStr = options.getIgnoreColumns();
+    if (ignoreStr == null || ignoreStr.isEmpty()) {
+      return Collections.emptyMap();
+    }
+    Map<String, Set<String>> ignore = new HashMap<>();
+    for (String tableColumns : ignoreStr.split(",")) {
+      int tableNameIndex = tableColumns.indexOf(':');
+      if (tableNameIndex == -1) {
+        continue;
+      }
+      String table = tableColumns.substring(0, tableNameIndex);
+      String columnStr = tableColumns.substring(tableNameIndex + 1);
+      Set<String> columns = new HashSet<>(Arrays.asList(columnStr.split(";")));
+      ignore.put(table, columns);
+    }
+    return ignore;
+  }
+
+  private static JdbcIO.ReadWithPartitions<Mutation, Long> getJdbcReader(
+      String table, Set<String> columnsToIgnore, JdbcToSpannerOptions options) {
+    return JdbcIO.<Mutation>readWithPartitions()
+        .withDataSourceConfiguration(getDataSourceConfiguration(options))
+        .withTable(table)
+        .withPartitionColumn(options.getPartitionColumn())
```

Comment:
PartitionColumn is an optional parameter, what happens if user does not pass it?